python/tvm/octo/utils/target_info.py

```diff
@@ -0,0 +1,156 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=invalid-name, wrong-import-position
+"""Utility functions for finding information about current device."""
+import os
+import re
+import sys
+import psutil
+import subprocess
+import shutil
+import tvm
+
+
+def get_llvm_target() -> tvm.target.Target:
+    """Extract fully featured llvm target for current device.
+
+    Returns
+    -------
+    target : tvm.target.Target
+        A TVM target that fully describes the current devices CPU.
+    """
+    # If we cant find llc, we wont be able to extract more information.
+    if shutil.which("llc") is None:
+        return "llvm"
+
+    # Get host information from llc
+    stream = subprocess.run(["llc", "--version"], stdout=subprocess.PIPE)
+    cpu_info = stream.stdout.decode("utf-8")
+
+    # Parse out cpu line
+    cpu = re.search("(?<=Host CPU: ).+", cpu_info).group(0)
+
+    # Next extract attribute string.
+    platform = sys.platform
+    # Linux
+    if platform in ["linux", "linux2"]:
+        output = subprocess.check_output("lscpu", shell=True).decode()
+        pattern = r"^([^:]+):\s+(.*)$"
```

Comment:
It might be good to give a comment to describe regexes like this one and the one above. They tend to be harder to understand when reading code